xonsh/jobs.py

```diff
@@ -199,6 +199,54 @@ def add_job(info):
         print_one_job(num)
 
 
+def clean_jobs():
+    """Clean up jobs for exiting shell
+
+    In non-interactive mode, kill all jobs.
+
+    In interactive mode, check for suspended or background jobs, print a
+    warning if any exist, and return False. Otherwise, return True.
+    """
+    jobs_clean = True
+    if builtins.__xonsh_env__['XONSH_INTERACTIVE']:
+        _clear_dead_jobs()
+        job_types = set()
+        for job in builtins.__xonsh_all_jobs__.values():
+            if job['bg']:
+                job_types.add('background')
+            elif job['status'] == 'stopped':
+                job_types.add('stopped')
+
+        if job_types:
+            if 'background' in job_types and 'stopped' in job_types:
+                type_str = 'background and stopped'
+            else:
+                type_str = job_types.pop()
+
+            if len(builtins.__xonsh_all_jobs__) > 1:
+                article = ''
+                job_str = 'jobs'
+            else:
+                article = 'a '
+                job_str = 'job'
+
+            print('xonsh: you have {art}{typ} {job}'.format(art=article,
+                                                            typ=type_str,
+                                                            job=job_str))
+            print('-'*5)
+            jobs([])
+            print('-'*5)
+            resp = input('Kill running jobs [y/n]? ')
+            if 'yes'.startswith(resp.lower()):
```

Comment:
Is the intention that an empty string (just hitting enter) will kill the jobs and quit?  Right now, it does.  If not, we'll need an explicit test for the empty string.  Or we could put this in a loop (while the input isn't either `'y'` or `'n'`, keep asking), but that feels a little painful.

Either way, we should probably indicate which of the options is the default by capitalizing it.

Maybe `Kill running jobs and exit [Y/n]?`